### PR TITLE
CR-1087409 Provide IOPS test host code for new validate kernels

### PR DESCRIPTION
**Pull request by mamin506**
What are added:
1. Support IOPS test case in new 'xbutil validate' flow.
2. Show CU name at the end of IOPS test result
    - Old flow, kernel name 'hello' and has 7 arguments.
    - New flow, kernel name 'verify' and has 1 argument.
3. Add internal API, xclOpenByBDF();

Test result:
U30
```
$xbutil validate -d 0
...
INFO: == Starting verify kernel test:
INFO: == verify kernel test PASSED
INFO: == Starting IOPS test:
Maximum IOPS: 94838 (verify)
INFO: == IOPS test PASSED
...
```
U50
```
...
INFO: == Starting verify kernel test: 
INFO: == verify kernel test PASSED
INFO: == Starting IOPS test: 
Maximum IOPS: 184314 (hello)
INFO: == IOPS test PASSED
...
```

Hi @vishnuchebrolu, I have one change in cmdlineparser.cpp. In before, for a toggle switch, the default value is true. I thought this is a bug. Please confirm. I have a toggle switch '-v' to turn verbose mode on/off.

### CODE REVIEW

**Comment by vishnuchebrolu** [comment]
@mamin506 , We had the following as part of commandline parser in Vitis Git Examples : 
https://gitenterprise.xilinx.com/SDxGitExamples/Vitis_Accel_Examples/blob/master/common/includes/cmdparser/cmdlineparser.cpp#L106
Can you explain why the default is changed to false?

File: src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp [hunk 1]
```diff
@@ -1191,7 +1193,12 @@ int xcldev::device::runTestCase(const std::string& py,
         auto device = pcidev::get_dev(m_idx);
         std::string bdf = boost::str(boost::format("%04x:%02x:%02x.%01x") % device->domain % device->bus % device->dev % device->func);
 
-        cmd = xrtTestCasePath + " " + xclbinPath + " -d " + bdf;
+        // Other new flow test cases are using OpenCL APIs, which can find device by BDF.
+        // If XRT and XCL APIs support open device by BDF, we can unify below two cmds.
+        if (py.find("xrt_iops_test") || py.find("xcl_iops_test"))
+            cmd = xrtTestCasePath + " " + xclbinPath + " -d " + std::to_string(m_idx) + " " + args;
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp:1199]: we have to support opening device by bdf, m_idx may point to a different device in iops test process.

**Review by maxzhen — CHANGES_REQUESTED** [review]

**Comment by mamin506** [comment]
> @mamin506 , We had the following as part of commandline parser in Vitis Git Examples :
> https://gitenterprise.xilinx.com/SDxGitExamples/Vitis_Accel_Examples/blob/master/common/includes/cmdparser/cmdlineparser.cpp#L106
> Can you explain why the default is changed to false?

In addSwitch(), the origin code is "cmd.isvalid = true". In parse(), it uses "pcmd->isvalid = true;". So, no matter the toggle switch applied or not, isValid() would always return true.

**Review by AShivangi — APPROVED** [review]
This looks good. 

@maxzhen, we don't have iops test support in xbutil2. Do you think we can skip adding this for the Beta release?

**Comment by maxzhen** [comment]
> 
> 
> This looks good.
> 
> @maxzhen, we don't have iops test support in xbutil2. Do you think we can skip adding this for the Beta release?

Yes, we don't need this for U30 Beta.

**Comment by mamin506** [comment]
@stsoe As per our discussion, add shim_int.h for an internal shim API xclOpenByBDF() for Linux PCIe platform.

File: src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp (hunk 1 above)
Comment (mamin506) [reply src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp]: Fixed.

**Review by maxzhen — APPROVED** [review]

**Review by stsoe — APPROVED** [review]

**Pull request merged**